higlass/tilesets.py

```diff
@@ -1,159 +1,176 @@
+        tileset_info=lambda: tileset_info(filepath, chromsizes),
+        tiles=lambda tids: tiles(filepath, tids, chromsizes=chromsizes),
+        datatype='vector',
+        **kwargs
+    )
 
 
-def mrmatrix(filepath, uuid=None):
-    import clodius.tiles.format as hgfo
-    import clodius.tiles.mrmatrix as hgmm
+def mrmatrix(filepath, uuid=None, **kwargs):
+    from clodius.tiles.mrmatrix import tileset_info, tiles
 
     f = h5py.File(filepath, "r")
 
     return Tileset(
         uuid=uuid,
-        tileset_info=lambda: hgmm.tileset_info(f),
-        tiles=lambda tile_ids: hgut.tiles_wrapper_2d(
-            tile_ids, lambda z, x, y: hgfo.format_dense_tile(hgmm.tiles(f, z, x, y))
+        tileset_info=lambda: tileset_info(f),
+        tiles=lambda tile_ids: tiles_wrapper_2d(
+            tile_ids, lambda z, x, y: format_dense_tile(tiles(f, z, x, y))
         ),
+        datatype='matrix',
+        **kwargs
     )
 
-import clodius.tiles.nplabels as ctnl
-import clodius.tiles.npvector as ctn
-import numpy as np
 
-def nplabels(labels_array, importances=None):
+def nplabels(labels_array, uuid=None, importances=None, **kwargs):
     """1d labels"""
+    from clodius.tiles import nplabels
+    from clodius.tiles import npvector
+
     return Tileset(
         uuid=uuid,
-        tileset_info=lambda: ctn.tileset_info(labels_array,
+        tileset_info=lambda: npvector.tileset_info(labels_array,
             bins_per_dimension=16),
-        tiles=lambda tids: ctnl.tiles_wrapper(labels_array,
-            tids, importances)
+        tiles=lambda tids: nplabels.tiles_wrapper(
+            labels_array, tids, importances),
+        datatype='linear-labels',
+        **kwargs
     )
 
-def h5labels(filename):
+
+def h5labels(filename, uuid=None, importances=None, **kwargs):
     f = h5py.File(filename, 'r')
-    return nplabels(f['labels'])
+    return nplabels(f['labels'], uuid, importances, **kwargs)
 
-def dfpoints(
-        df: pd.DataFrame,
-        x_col: str,
-        y_col: str,
-        uuid: str = None):
+
+def dfpoints(df, x_col, y_col, uuid=None, **kwargs):
     """
     Generate a tileset that serves 2d labelled points from a pandas
     dataframe.
-    Parameters:
-    -----------
-    df: The dataframe containining the data
-    x_col: The name of the column containing the x-coordinates
-    y_col: The name of the column containing the y-coordinates
-    uuid: The uuid of this tileset
-    Returns:
-    --------
+
+    Parameters
+    ----------
+    df: :class:`pandas.DataFrame`
+        The dataframe containining the data
+    x_col: str
+        The name of the column containing the x-coordinates
+    y_col: str
+        The name of the column containing the y-coordinates
+    uuid: str
+        The uuid of this tileset
+
+    Returns
+    -------
     A tileset capapble of serving tiles from this dataframe.
+
     """
-    import clodius.tiles.points as hgpo
+    from clodius.tiles.points import tileset_info, tiles, format_data
```

Comment:
Could we use the import style you introduced in the previous function throughout?

```
    from clodius.tiles import nplabels
```

Reply:
I don't see the reason for the inner function imports because it adds an indirection step and the imports are immediately visible above the code that uses them.

For global imports I normally wouldn't do this either because of possible namespace clashes.